randovania/games/am2r/exporter/patch_data_factory.py

```diff
@@ -1,10 +1,200 @@
+                                        "Jump into enemies to inflict massive damage")
+    info["Varia Suit"] = AM2RItemInfo("sItemVariaSuit", def_speed, "Varia Suit acquired", "Reduces Damage taken by 50%")
+    info["Space Jump"] = AM2RItemInfo("sItemSpaceJump", def_speed, "Space Jump acquired",
+                                      "Jump continuously in the air")
+    info["Speed Booster"] = AM2RItemInfo("sItemSpeedBooster", 0.5, "Speed Booster acquired",
+                                         "Run continuously to begin Speed Boosting")
+    info["Hi-Jump"] = AM2RItemInfo("sItemHijump", def_speed, "Hi-Jump acquired", "Maximum Jump height increased")
+    info["Gravity Suit"] = AM2RItemInfo("sItemGravitySuit", def_speed, "Gravity Suit acquired",
+                                        "Liquid friction eliminated - Improved Defense")
+    info["Charge Beam"] = AM2RItemInfo("sItemChargeBeam", def_speed, "Charge Beam acquired",
+                                       "Hold | to charge your beam")
+    info["Ice Beam"] = AM2RItemInfo("sItemIceBeam", def_speed, "Ice Beam acquired",
+                                    "Instantly freeze enemies. Extra damage")
+    info["Wave Beam"] = AM2RItemInfo("sItemWaveBeam", def_speed, "Wave Beam acquired",
+                                     "Beams can now go through solid objects")
+    info["Spazer Beam"] = AM2RItemInfo("sItemSpazerBeam", def_speed, "Spazer Beam acquired",
+                                       "Beams now have a wider range")
+    info["Plasma Beam"] = AM2RItemInfo("sItemPlasmaBeam", def_speed, "Plasma Beam acquired", "Beams can pierce enemies")
+    info["Morph Ball"] = AM2RItemInfo("sItemMorphBall", 0.1666, "Morph Ball acquired", "Roll around in a ball")
+    info["Nothing"] = AM2RItemInfo("sItemUnknown", 0.3333, "Nothing acquired", "A useless item")
+    info["EnergyTransferModule"] = info["Nothing"]
+    info["Unknown Item"] = AM2RItemInfo("sItemUnknown", 0.3333, "Got Unknown Item", "A mysterious item")
+
+    return info
 
 
 class AM2RPatchDataFactory(BasePatchDataFactory):
+    _EASTER_EGG_SHINY_MISSILE = 1024
+
+    def _create_pickups_dict(self, pickup_list, item_info, rng: Random):
+        pickup_map_dict = {}
+        for pickup in pickup_list:
+            quantity = pickup.conditional_resources[0].resources[0][1]
+            object_name = self.game.region_list.node_from_pickup_index(pickup.index).extra["object_name"]
+            pickup_map_dict[object_name] = {
+                "sprite_details": {
+                    "name": item_info[pickup.model.name].sprite_name,
+                    "speed": item_info[pickup.model.name].sprite_speed
+                },
+                "item_effect": pickup.original_pickup.name,
+                "quantity": quantity,
+                "text": {
+                    "header": item_info[pickup.model.name].text_header,
+                    "description": item_info[pickup.name].text_desc.replace("[capacity]", str(quantity)).replace(
```

Comment:
Check python's `str.format`.

But actually look how Echoes' did the [original memo data](https://github.com/randovania/randovania/blob/08d50edff96f805a8762582d324366f137a108f7/randovania/games/prime2/pickup_database/memo_data.json) feature. It has very different text for each pickup, but is still dynamic with ammo count. And you can just grab the text from the pickup details!

Reply:
Oh, so by using memo data,  the text / description gets put into the pickup details itself?

Reply:
The `collection_text` is built using the memo data you pass to the PickupExporter.
The description is currently hard coded, but I doubt you'll use it.